GeoLib/AnalyticalGeometry.h

```diff
@@ -299,6 +299,10 @@ bool parallel(MathLib::Vector3 v, MathLib::Vector3 w);
 bool lineSegmentIntersect (const GeoLib::Point& a, const GeoLib::Point& b,
         const GeoLib::Point& c, const GeoLib::Point& d, GeoLib::Point& s);
 
+std::vector<MathLib::Point3d>
+lineSegmentIntersect2d(MathLib::Point3d const& a, MathLib::Point3d const& b,
```

Comment:
In principle it is possible to use `LineSegment` objects here. The main reason I want to postpone this is that `GeoLib::Point`s are used for the definition of  a `LineSegment`. I want to use this algorithm also for line segments build from `MeshLib::Node`s. At the moment `LineSegment` is only for testing purposes.